### PR TITLE
Add FeatureCollection mutate tests (not failing)

### PR DESCRIPTION
**Pull request by DenisCarriere**
# Add test cases for `@turf/transform-scale`

- [x] Adding failing tests for https://github.com/Turfjs/turf/issues/895
- [x] Test mutating FeatureCollection (not the issue)
- [x] Fix issue

## Issue

On (Multi)Polygon the first & last coordinate gets repeated when using `coordEach` (not sure if this was intentional in `coordEach`).

## Before

image

## After

image

### CODE REVIEW

File: packages/turf-transform-scale/index.js [hunk 1]
```diff
@@ -59,18 +61,24 @@ module.exports = function (geojson, factor, origin, mutate) {
...
     origin = defineOrigin(geojson, origin);
 
     // Shortcut no-scaling
     if (factor === 1 || isPoint) return geojson;
 
     // Scale each coordinate
-    coordEach(geojson, function (coord) {
+    coordEach(geojson, function (coord, coordIndex, featureIndex, featureSubIndex) {
+        // Ignore scaling on first coordinate of Polygons (Issue #895)
+        if (isPolygon && mutate === true && featureSubIndex === 0) return;
```
Comment (DenisCarriere) [inline packages/turf-transform-scale/index.js]: @stebogit Might want to add this type of behavior in a few of the `transform` modules.

File: packages/turf-transform-scale/index.js [hunk 2]
```diff
@@ -59,18 +61,24 @@ module.exports = function (geojson, factor, origin, mutate) {
...
+    var isPoint = geomType === 'Point';
+    var isPolygon = geomType === 'Polygon' || geomType === 'MultiPolygon';
     origin = defineOrigin(geojson, origin);
 
     // Shortcut no-scaling
     if (factor === 1 || isPoint) return geojson;
 
     // Scale each coordinate
-    coordEach(geojson, function (coord) {
+    coordEach(geojson, function (coord, coordIndex, featureIndex, featureSubIndex) {
```
Comment (stebogit) [inline packages/turf-transform-scale/index.js]: I noticed `coordIndex` and `featureSubIndex` are always equals, throughout the entire test suite, and `featureIndex` is always `0`. 
Inside coordEach this never logs:
```js
if (coordIndex !== featureSubIndex || featureIndex > 0) console.log('Woah!');
```
Shouldn't the indices change at least for multi features?

File: packages/turf-transform-scale/index.js (hunk 1 above)
Comment (stebogit) [inline packages/turf-transform-scale/index.js]: @DenisCarriere it looks ok, and clearly it does work. However I can not understand why we don't want to scale all the points of the ring(s) of a polygon, whether we clone it or not.
I really can't wrap my head around it... 🤔

File: packages/turf-transform-scale/index.js [hunk 3]
```diff
@@ -59,18 +61,24 @@ module.exports = function (geojson, factor, origin, mutate) {
  * @param {Feature|Geometry} geojson GeoJSON Feature/Geometry
  * @param {number} factor of scaling, positive or negative values greater than 0
  * @param {string|Geometry|Feature<Point>|Array<number>} [origin="centroid"] Point from which the scaling will occur (string options: sw/se/nw/ne/center/centroid)
+ * @param {boolean} [mutate=false] allows GeoJSON input to be mutated
```
Comment (stebogit) [inline packages/turf-transform-scale/index.js]: Here `mutate` does not actually define if the feature is going to be mutated or not, that was handled before.
I can't find a good description though 😅

**Comment by stebogit** [comment]
@DenisCarriere I found out the real issue: as you [said](https://github.com/Turfjs/turf/issues/895#issuecomment-321849798)
> The scaling happens twice on the first & last coordinate position (for some reason...)

because when `@turf/hex-grid` created the polygons the last vertex of the ring was set as reference to the first one, instead of cloning it.
screen shot 2017-08-13 at 1 58 58 am

I modified (locally) that line and all seems to be good with the original `transform-scale` code (but of course this branch still fails until we update `@turf/hex-grid`).

**Comment by DenisCarriere** [comment]
> because when @turf/hex-grid created the polygons the last vertex of the ring was set as reference to the first one, instead of cloning it.

Ahh... that makes sense! Was wondering where that issue came from.

File: packages/turf-transform-scale/index.js [hunk 4]
```diff
@@ -46,39 +46,33 @@ module.exports = function (geojson, factor, origin, mutate) {
...
-    return scale(geojson, factor, origin, mutate);
+    return scale(geojson, factor, origin);
 };
 
 /**
  * Scale Feature/Geometry
  *
  * @private
- * @param {Feature|Geometry} geojson GeoJSON Feature/Geometry
+ * @param {GeoJSON} geojson GeoJSON Feature/Geometry
```
Comment (DenisCarriere) [inline packages/turf-transform-scale/index.js]: GeoJSON means it handles FeatureCollection & GeometryCollection.

This doesn't?

File: packages/turf-transform-scale/index.js [hunk 5]
```diff
@@ -56,14 +58,14 @@ module.exports = function (geojson, factor, origin, mutate) {
...
  * @param {number} factor of scaling, positive or negative values greater than 0
  * @param {string|Geometry|Feature<Point>|Array<number>} [origin="centroid"] Point from which the scaling will occur (string options: sw/se/nw/ne/center/centroid)
- * @returns {Feature|Geometry} scaled GeoJSON Feature/Geometry
+ * @returns {GeoJSON} scaled GeoJSON object
```
Comment (DenisCarriere) [inline packages/turf-transform-scale/index.js]: If the input is Geometry it outputs a scaled Geometry, we know the outputs/inputs can ONLY be Feature or Geometry.

**Pull request merged**